redis/src/cluster.rs

```diff
@@ -730,6 +727,9 @@ where
                         SingleNodeRoutingInfo::SpecificNode(route) => {
                             self.get_connection(&mut connections, route)?
                         }
+                        SingleNodeRoutingInfo::RandomPrimary => {
```

Comment:
1. The code for `RandomPrimary` is correct
2. The `RoutingInfo::SingleNode(SingleNodeRoutingInfo::Random)` was not changed (replaced the old code with a call to `Route::new_random_primary()`
 
Having said that, there is no bug on that line, however there can be an optimization: for example we could add a check to see whether the cmd is readonly, and in case it does, we could allow "read-from-replica" instead of "SlotAddr::Master`.

If you think that this is correct, we could open a new bug and fix it in a different PR (I prefer not to mix changes in a single PR)